### ISSUE TITLE
Some command palette commands do nothing

### ISSUE THREAD

**Issue by Kregap** [godotengine/godot#78981]
### Godot version

v4.1.rc2.official [46424488e]

### System information

Windows 11

### Issue description

Selecting some of the commands from command palette do nothing. Seems like commands that should open a popup/window fail, while other commands work. Selecting faulty commands from menu directly work.

https://github.com/godotengine/godot/assets/15647621/41deb0b8-3896-4d08-bc10-92d900254de7

### Steps to reproduce

1. Open commands palette (Ctrl+Shift+P).
2. Select "Editor Settings" from the list.
3. Confirm command selection with either return keyboard key or by clicking on "OK" button.
**Expected behaviour**
Command palette popup hides.
Editor Settings popup opens.
**Actual behaviour**
Command palette popup hides but Editor Settings popup does not open.

### Minimal reproduction project

N/A

**Comment by Sauermann** [comment]
I was able to bisect this to #76719, which was also ported to 3.x.

**Comment by passivestar** [comment]
Same problem.
Godot v4.1.1.stable - macOS 13.5.2

Trying to execute "Project Settings", "Export", "New Scene", etc has no effect

**Comment by SirLich** [comment]
Just a heads-up that I am actively trying to solve this bug. I think I've got some good guesses, but until I can attach a debugger it will be tricky. 

I'm quite suspicious of this snippet:
```cpp
void EditorCommandPalette::_confirmed() {
	TreeItem *selected_option = search_options->get_selected();
	String command_key = selected_option != nullptr ? selected_option->get_metadata(0) : "";
	if (!command_key.is_empty()) {
		hide();
		execute_command(command_key);
	}
}
```

**Comment by mcdz** [comment]
Same issue in v4.2 dev 5

**Comment by kleonc** [comment]
> Same issue in v4.2 dev 5

@mcdz `v4.2.dev5` was released 9 days ago, #82203 (fixing this issue) was merged 4 days ago and hence is not included in `4.2.dev5`. It will be included in the next version.
